#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/RectangleArena.proto"

WorldInfo {
}
Viewpoint {
  orientation 0.2567754988874531 0.6693647675123979 -0.6971493033663398 1.00606874507109
  position -0.6322043272556068 -0.10636516125271522 0.4987554132812675
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 2 2
  floorAppearance PBRAppearance {
    roughness 1
    roughnessMap ImageTexture {
      url [
        "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/textures/parquetry/mosaic_parquetry_roughness.jpg"
      ]
    }
    metalness 0
    normalMap ImageTexture {
      url [
        "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/textures/parquetry/mosaic_parquetry_normal.jpg"
      ]
    }
    occlusionMap ImageTexture {
      url [
        "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/textures/parquetry/mosaic_parquetry_occlusion.jpg"
      ]
    }
  }
}
Solid {
  translation 0.45 -0.73 0.05
  rotation 0 0 -1 -0.13090030610042527
  children [
    DEF obstacle1 Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.1 0.1
      }
    }
  ]
  boundingObject USE obstacle1
  locked TRUE
}
Solid {
  translation 0.666748 -0.198009 0.05
  rotation 0 0 -1 -1.3089953071795861
  children [
    DEF obstacle2 Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.2 0.1 0.1
      }
    }
  ]
  name "solid(1)"
  boundingObject USE obstacle2
  locked TRUE
}
Solid {
  translation 0.36358 0.381911 0.05
  rotation 0 0 -1 -1.047195307179586
  children [
    DEF obstacle3 Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.12 0.1 0.1
      }
    }
  ]
  name "solid(3)"
  boundingObject USE obstacle3
  locked TRUE
}
Solid {
  translation -0.380764 0.661551 0.05
  rotation 0 0 1 2.09439
  children [
    DEF obstacle4 Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.12 0.1 0.1
      }
    }
  ]
  name "solid(4)"
  boundingObject USE obstacle4
  locked TRUE
}
Solid {
  translation -0.545793 -0.0756207 0.05
  rotation 0 0 1 1.83259
  children [
    DEF obstacle5 Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.3 0.1
      }
    }
  ]
  name "solid(5)"
  boundingObject USE obstacle5
  locked TRUE
}
Solid {
  translation -0.356546 -0.527723 0
  rotation 0 0 1 2.8797896938995744
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.03 0.51 0.0005
      }
    }
  ]
  name "solid(6)"
  locked TRUE
}
Solid {
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
        roughness 1
        metalness 0
      }
      geometry Mesh {
        url [
          "../path/outer_path.obj"
        ]
      }
    }
  ]
  name "solid(2)"
  boundingObject Mesh {
    url [
      "../path/outer_path.obj"
    ]
  }
  locked TRUE
}
Solid {
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
        roughness 1
        metalness 0
      }
      geometry Mesh {
        url [
          "../path/inner_path.obj"
        ]
      }
    }
  ]
  name "solid(7)"
  boundingObject Mesh {
    url [
      "../path/inner_path.obj"
    ]
  }
  locked TRUE
}
Robot {
  translation -0.1576388688693435 -0.5683217659374444 0.028810428556339848
  rotation -0.030141048227757112 -0.6334927347439053 -0.7731613494209472 0.12304904967377091
  children [
    DEF cam Solid {
      translation 0.049 0 0.019
      rotation 0 1 0 5.307156870259904e-06
      children [
        DEF camera Camera {
          fieldOfView 2.5
          width 640
          height 640
        }
        Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.005
            radius 0.005
          }
        }
      ]
      name "solid(1)"
    }
    Pose {
      children [
        DEF Body Shape {
          appearance PBRAppearance {
            baseColor 0 0.3 1
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.05
            radius 0.05
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 1743.480936618949
        axis 0 1 0
        anchor 0 0.05 -0.01
      }
      device [
        RotationalMotor {
          name "motor_1"
        }
        PositionSensor {
          name "enc_1"
        }
      ]
      endPoint Solid {
        translation 0 0.05 -0.01
        rotation -0.036419193906410435 -0.7066364976692854 0.7066388769922679 3.214398618825471
        children [
          DEF Wheel Shape {
            appearance PBRAppearance {
              baseColor 0 0 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.005
              radius 0.019
            }
          }
        ]
        boundingObject USE Wheel
        physics Physics {
        }
        linearVelocity 0.10803447706246933 -0.010667004692099245 8.238377093576583e-08
        angularVelocity 0.5599139782716883 5.672141413926394 0.11117564593721391
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 1724.7052331481773
        axis 0 1 0
        anchor 0 -0.05 -0.01
      }
      device [
        RotationalMotor {
          name "motor_2"
        }
        PositionSensor {
          name "enc_2"
        }
      ]
      endPoint Solid {
        translation 0 -0.05 -0.01
        rotation 0.010300508201475765 0.7070680776218982 -0.7070704583977927 3.120992435091503
        children [
          DEF Wheel Shape {
            appearance PBRAppearance {
              baseColor 0 0 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.005
              radius 0.019
            }
          }
        ]
        boundingObject USE Wheel
        physics Physics {
        }
        linearVelocity 0.11886344521053664 -0.011735008591241962 1.0263561789855687e-06
        angularVelocity 0.6191586971718054 6.269825141419402 0.1112241361128321
      }
    }
  ]
  boundingObject USE Body
  physics Physics {
  }
  controller "main"
  linearVelocity 0.11344083170092124 -0.011285312746747073 4.7170694516849096e-07
  angularVelocity -8.345547176477578e-06 1.6929300159604278e-05 0.10881281702356779
}
